src/compiler/checker.ts

```diff
@@ -21465,7 +21465,16 @@ namespace ts {
         }
 
         function checkPostfixUnaryExpression(node: PostfixUnaryExpression): Type {
-            const operandType = checkExpression(node.operand);
+            let operandType: Type;
+            const symbol = (<Identifier>node.operand).escapedText ? getResolvedSymbol(node.operand as Identifier) : undefined;

```

Comment:
Use `isEntityNameExpression` and not `left.operand.escapedText` (since checking `a.b++` is also desirable. Also, a test for `a.b++` would be good.